src/systems/renderer.js

```diff
@@ -83,3 +96,59 @@ module.exports.System = registerSystem('renderer', {
     }
   }
 });
+
+// Custom A-Frame sort functions.
+// Variations of Three.js default sort orders here:
+// https://github.com/mrdoob/three.js/blob/ebbaecf9acacf259ea9abdcba7b6fb25cfcea2ab/src/renderers/webgl/WebGLRenderLists.js#L1
+// See: https://github.com/aframevr/aframe/issues/5332
+
+// Default sort for opaque objects:
+// - respect groupOrder & renderOrder settings
+// - sort front-to-back by z-depth from camera (this should minimize overdraw)
+// - otherwise leave objects in default order (object tree order)
+
+function sortOpaqueDefault (a, b) {
```

Comment:
Can we find a bit more descriptive names for these functions? I had suggested `sortOpaqueObjectsFirst` but not sure if good enough. can suggest other names.

Reply:
If we can make this function names a bit more descriptive this is almost ready to merge. Thanks!

Reply:
Sorry for slow response on this.

I just pushed a change that updated:
`sortOpaqueDefault` -> `sortSpatialFrontToBack`
`sortTransparentDefault` -> `sortRenderOrderOnly`
`sortTransparentSpatial` -> `sortSpatialBackToFront`

This describes what the functions implement rather than what they are used for (glossing over the detail that the spatial sorts also sort by `renderOrder` first, in the interest of keeping the function names at a sensible length).